components/Snackbar/src/private/MDCSnackbarOverlayView.m

```diff
@@ -246,51 +250,41 @@ - (void)setSnackbarView:(MDCSnackbarMessageView *)snackbarView {
-    BOOL fullWidth = UI_USER_INTERFACE_IDIOM() != UIUserInterfaceIdiomPad;
+    CGFloat leftMargin = sideMargin;
+    CGFloat rightMargin = sideMargin;
 
     UIView *container = self.containingView;
 
     if (snackbarView) {
       [container addSubview:snackbarView];
 
-      // Pin the snackbar to the bottom of the screen.
+      // Pin the Snackbar to the bottom of the screen.
       [snackbarView setTranslatesAutoresizingMaskIntoConstraints:NO];
 
-      if (fullWidth) {
-        CGFloat leftMargin = sideMargin;
-        CGFloat rightMargin = sideMargin;
-
-#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
-        if (@available(iOS 11.0, *)) {
-          leftMargin += self.safeAreaInsets.left;
-          rightMargin += self.safeAreaInsets.right;
-        }
-#endif
-
-        [container addConstraint:[NSLayoutConstraint constraintWithItem:snackbarView
-                                                              attribute:NSLayoutAttributeLeading
-                                                              relatedBy:NSLayoutRelationEqual
-                                                                 toItem:container
-                                                              attribute:NSLayoutAttributeLeading
-                                                             multiplier:1.0
-                                                               constant:leftMargin]];
-
-        [container addConstraint:[NSLayoutConstraint constraintWithItem:snackbarView
-                                                              attribute:NSLayoutAttributeTrailing
-                                                              relatedBy:NSLayoutRelationEqual
-                                                                 toItem:container
-                                                              attribute:NSLayoutAttributeTrailing
-                                                             multiplier:1.0
-                                                               constant:-1 * rightMargin]];
-      } else {
-
-        [container addConstraint:[NSLayoutConstraint constraintWithItem:snackbarView
-                                                              attribute:NSLayoutAttributeCenterX
-                                                              relatedBy:NSLayoutRelationEqual
-                                                                 toItem:container
-                                                              attribute:NSLayoutAttributeCenterX
-                                                             multiplier:1.0
-                                                               constant:0]];
+      BOOL isRegularWidth =
+          self.traitCollection.horizontalSizeClass == UIUserInterfaceSizeClassRegular;
+      BOOL isRegularHeight =
+          self.traitCollection.verticalSizeClass == UIUserInterfaceSizeClassRegular;
+      if (isRegularWidth && isRegularHeight) {
```

Comment:
If you want to make it easier to unit test, consider refactoring this check into a method that can be (subclassed and) overridden (or partially mocked).

For example, 

```objectivec
- (BOOL)permitsLeadingPresentation {
  // Normally do the trait collection checks here
}

- (void)setSnackbarView:(MDCSnackbarMessageView *)snackbarView {
  ...
  if ([self permitsLeadingPresentation]) {
    ...
  }
}
```

With that, you could hopefully test/verify the behavior of `setSnackbarView:` without requiring actual trait collections to exist.

Reply:
The other option is to file an Issue in the backlog and come back to it later.

Reply:
Ah, just saw you referenced #3823